src/vs/base/browser/performance.ts

```diff
@@ -0,0 +1,231 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+export namespace inputLatency {
+
+	// Measurements are recorded in typed arrays with a fixed buffer length to their own impact on
+	// input latency.
+	const enum Constants {
+		bufferLength = 256
+	}
+	const measurementsKeydown = new Float32Array(Constants.bufferLength);
+	const measurementsInput = new Float32Array(Constants.bufferLength);
+	const measurementsRender = new Float32Array(Constants.bufferLength);
+	const measurementsInputLatency = new Float32Array(Constants.bufferLength);
+	let measurementsCount = 0;
+
+	// The state of each event, this helps ensure the integrity of the measurement and that
+	// something unexpected didn't happen that could skew the measurement.
+	const enum EventPhase {
+		Before = 0,
+		InProgress = 1,
+		Finished = 2
+	}
+	const state = {
+		keydown: EventPhase.Before,
+		input: EventPhase.Before,
+		render: EventPhase.Before,
+		selection: EventPhase.Before
+	};
+
+	/**
+	 * Mark the start of the keydown event.
+	 */
+	export function markKeydownStart() {
+		performance.mark('inputlatency/start');
```

Comment:
Using `performance.mark` might interact negatively with `src/vs/base/common/performance.js`, but I don't know enough about the browser API to tell for sure.

Reply:
Let's add @jrieken to confirm that there are no negative interactions here.